tests/dialects/test_tsql.py

```diff
@@ -389,7 +389,62 @@ def test_types_bin(self):
             },
         )
 
+    def test_ddl(self):
+        sql = """CREATE TABLE #mytemp (a INTEGER,  b CHAR(2), c TIME(4), d FLOAT(24));"""
+        self.validate_all(
+            " ".join(sql.split()),
```

Comment:
This doesn't seem correct. You can do:

```python
self.validate_identity(
    "CREATE TABLE #mytemp (a INTEGER, b CHAR(2), c TIME(4), d FLOAT(24))"
)
```

Reply:
I had another test for spark/databricks, but I removed that from this PR. I'll clean this for now.

Reply:
Also TIME(4) gets mapped to TIMESTAMP(4)

Reply:
I removed the weird join.split() which was used for the other match.